native/NativeSampleApps/SmartSyncExplorer/src/com/salesforce/samples/smartsyncexplorer/sync/ContactSyncAdapter.java

```diff
@@ -63,12 +64,14 @@ public ContactSyncAdapter(Context context, boolean autoInitialize,
     public void onPerformSync(Account account, Bundle extras, String authority,
                               ContentProviderClient provider, SyncResult syncResult) {
         final boolean syncDownOnly = extras.getBoolean(SYNC_DOWN_ONLY, false);
-        final SalesforceSDKManager sdkManager = SalesforceSDKManager.getInstance();
+        final SmartStoreSDKManager sdkManager = SmartStoreSDKManager.getInstance();
         final UserAccountManager accManager = sdkManager.getUserAccountManager();
         if (sdkManager.isLoggingOut() || accManager.getAuthenticatedUsers() == null) {
             return;
         }
         if (account != null) {
+            // Setup schema if needed
+            sdkManager.setupUserStoreFromDefaultConfig();
```

Comment:
Shouldn't the setup happen from the loader? Seems like an anti-pattern to trigger this from the sync adapter.